File: app/src/main/java/com/do_big/diginotes/data/NoteRepository.java
package com.do_big.diginotes.data;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.do_big.diginotes.model.Note;
import com.do_big.diginotes.utils.NoteRoomDataBase;

import java.util.List;

public class NoteRepository {
    private final NoteDao noteDao;
    private final LiveData<List<Note>> allNotes;

    public NoteRepository(Application application) {
        NoteRoomDataBase noteRoomDataBase = NoteRoomDataBase.getINSTANCE(application);
        noteDao = noteRoomDataBase.noteDao();
        allNotes = noteDao.getNotes();
    }

    public LiveData<List<Note>> getAllNotes() {
        return allNotes;
    }

    public void insert(Note note) {
        NoteRoomDataBase.databaseWriterExecutor.execute(new Runnable() {
            @Override
            public void run() {
                noteDao.insertNote(note);
            }
        });
    }

    public LiveData<Note> get(long id) {
        return noteDao.get(id);
    }

    public void update(Note note) {
        NoteRoomDataBase.databaseWriterExecutor.execute(new Runnable() {
            @Override
            public void run() {
                noteDao.update(note);
            }
        });
    }

    public void delete(Note note) {
        NoteRoomDataBase.databaseWriterExecutor.execute(new Runnable() {
            @Override
            public void run() {
                noteDao.delete(note);
            }
        });
    }

}

File: app/src/main/java/com/do_big/diginotes/data/NoteRepository.kt
package com.do_big.diginotes.data

import android.app.Application
import androidx.lifecycle.LiveData
import com.do_big.diginotes.model.Note
import com.do_big.diginotes.utils.NoteRoomDataBase
import com.do_big.diginotes.utils.NoteRoomDataBase.Companion.getINSTANCE

class NoteRepository(application: Application) {
    private val noteDao: NoteDao

    @JvmField
    val allNotes: LiveData<List<Note>>

    init {
        val noteRoomDataBase = getINSTANCE(application)
        noteDao = noteRoomDataBase!!.noteDao()
        allNotes = noteDao.notes
    }

    fun insert(note: Note?) {
        NoteRoomDataBase.databaseWriterExecutor.execute { noteDao.insertNote(note) }
    }

    fun get(id: Long): LiveData<Note> {
        return noteDao[id]
    }

    fun update(note: Note?) {
        NoteRoomDataBase.databaseWriterExecutor.execute { noteDao.update(note) }
    }

    fun delete(note: Note?) {
        NoteRoomDataBase.databaseWriterExecutor.execute { noteDao.delete(note) }
    }
}
